lib/node_modules/@stdlib/blas/ext/base/wasm/dnansumkbn2/test/test.main.js

```diff
@@ -0,0 +1,191 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var Float64Array = require( '@stdlib/array/float64' );
+var dnansumkbn2 = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is an object', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof dnansumkbn2, 'object', 'main export is an object' );
+	t.end();
+});
+
+tape( 'the function has an arity of 3', function test( t ) {
+	t.strictEqual( dnansumkbn2.main.length, 3, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function calculates the sum of absolute values', function test( t ) {
+	var x;
+	var v;
+	var i;
+
+	x = new Float64Array( [ 1.0, -2.0, NaN, 5.0, 0.0, 3.0, NaN, -3.0, 3.0 ] );
+	v = dnansumkbn2.main( x.length, x, 1 );
+	t.strictEqual( v, 7.0, 'returns expected value' );
+
+	x = new Float64Array( [ 1.0, -2.0, NaN, 5.0, 0.0, 3.0 ] );
+	v = dnansumkbn2.main( x.length, x, 1 );
+	t.strictEqual( v, 7.0, 'returns expected value' );
+
+	x = new Float64Array( [ 4.0, NaN ] );
+	v = dnansumkbn2.main( x.length, x, 1 );
+	t.strictEqual( v, 4.0, 'returns expected value' );
+
+	x = new Float64Array( [ 1.0, 1.0e100, NaN, -1.0e100 ] );
+	v = dnansumkbn2.main( x.length, x, 1 );
+	t.strictEqual( v, 1.0, 'returns expected value' );
+
+	x = new Float64Array( 1e3 );
+	for ( i = 0; i < 1e3; i++ ) {
+		x[ i ] = i + 1;
+	}
+	v = dnansumkbn2.main( x.length, x, 1 );
+	t.strictEqual( v, 500500.0, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `0.0`', function test( t ) {
+	var x;
+	var v;
+
+	x = new Float64Array( [ 1.0, NaN, -4.0, 5.0, 3.0 ] );
+
+	v = dnansumkbn2.main( 0, x, 1 );
+	t.strictEqual( v, 0.0, 'returns expected value' );
+
+	v = dnansumkbn2.main( -1, x, 1 );
+	t.strictEqual( v, 0.0, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'if provided an `N` parameter equal to `1`, the function returns the first element', function test( t ) {
+	var x;
+	var v;
+
+	x = new Float64Array( [ 1.0, NaN, -4.0, 5.0, 3.0 ] );
+
+	v = dnansumkbn2.main( 1, x, 1 );
+	t.strictEqual( v, 1.0, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function supports a `stride` parameter', function test( t ) {
+	var x;
+	var v;
+	var N;
+
+	x = new Float64Array([
+		1.0,  // 0
+		2.0,
+		2.0,  // 1
+		-7.0,
+		NaN, // 2
```

Comment:
```suggestion
		NaN,  // 2
```

Alignment. Attention to detail.